### PR TITLE
all: Remove symlink support on Windows, SymlinksEnabled config

### PR DESCRIPTION
**Pull request by calmh**
After this change,

- Symlinks on Windows are always unsupported. Sorry.

- Symlinks are always enabled on other platforms. They are just a small
  file like anything else. There is no need to special case them. If you
  don't want to sync some symlinks, ignore them.

- The protocol doesn't differentiate between different "types" of
  symlinks. If that distinction ever does become relevant the individual
  devices can figure it out by looking at the destination when they
  create the link.

It's backwards compatible in that all the old symlink types are still
understood to be symlinks, and the new SYMLINK type is equivalent to the
old SYMLINK_UNKNOWN which was always a valid way to do it.

### Testing

Unit tests and existing integration tests on symlinks still work. For me. Lets see the builders.

### Documentation

Spec should be updated to deprecate the two old symlink types.

### CODE REVIEW

File: lib/protocol/bep.proto [hunk 1]
```diff
@@ -111,11 +111,11 @@ message FileInfo {
...
-    FILE              = 0 [(gogoproto.enumvalue_customname) = "FileInfoTypeFile"];
-    DIRECTORY         = 1 [(gogoproto.enumvalue_customname) = "FileInfoTypeDirectory"];
-    SYMLINK_FILE      = 2 [(gogoproto.enumvalue_customname) = "FileInfoTypeSymlinkFile"];
-    SYMLINK_DIRECTORY = 3 [(gogoproto.enumvalue_customname) = "FileInfoTypeSymlinkDirectory"];
-    SYMLINK_UNKNOWN   = 4 [(gogoproto.enumvalue_customname) = "FileInfoTypeSymlinkUnknown"];
+    FILE                         = 0 [(gogoproto.enumvalue_customname) = "FileInfoTypeFile"];
+    DIRECTORY                    = 1 [(gogoproto.enumvalue_customname) = "FileInfoTypeDirectory"];
+    DEPRECATED_SYMLINK_FILE      = 2 [(gogoproto.enumvalue_customname) = "FileInfoTypeDeprecatedSymlinkFile"];
+    DEPRECATED_SYMLINK_DIRECTORY = 3 [(gogoproto.enumvalue_customname) = "FileInfoTypeDeprecatedSymlinkDirectory"];
+    SYMLINK                      = 4 [(gogoproto.enumvalue_customname) = "FileInfoTypeSymlink"];
```
Comment (AudriusButkevicius) [inline lib/protocol/bep.proto]: int32 old_field = 6 [deprecated=true];

**Comment by AudriusButkevicius** [comment]
We should probably weed out and update the catalog entries on first scan not to have the deprecated fields, so that we could just drop the fields at some point in the future.

**Comment by AudriusButkevicius** [comment]
@st-review lgtm

**Comment by st-review** [comment]
@AudriusButkevicius: Noted! Need another LGTM or explicit merge command.

**Comment by AudriusButkevicius** [comment]
needs docs updates.

**Comment by ProactiveServices** [comment]
I'll leave the reference to symlinks in the [specs](https://github.com/syncthing/docs/blob/master/specs/bep-v1.rst) to someone more familiar with the effect of the changes.

**Comment by calmh** [comment]
Docs updated, a bit more drastically than your change @ProactiveServices  :)

**Comment by calmh** [comment]
@st-review merge it

**Comment by st-review** [comment]
@calmh: Merge failed:

```
$ git fetch -f origin refs/pull/3962/head:pr-3962
$ git fetch -f origin master:orig/master
From github.com:syncthing/syncthing
   e14741a..9fda964  master     -> orig/master
 * [new tag]         v0.14.22   -> v0.14.22
   fdcbd54..9fda964  master     -> origin/master
$ git reset --hard
HEAD is now at fdcbd54 lib/events: Be more resilient against dropping events (fixes #3952)
$ git checkout master
Already on 'master'
Your branch is behind 'origin/master' by 3 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
$ git reset --hard orig/master
HEAD is now at 9fda964 lib/events: Make test even less timing dependent
$ git clean -fxd
$ git merge --squash --no-commit pr-3962
Removing lib/symlinks/targets.go
Removing lib/symlinks/symlink_windows.go
Removing lib/symlinks/symlink_unix.go
Removing lib/symlinks/empty_test.go
Auto-merging lib/model/rwfolder.go
Auto-merging lib/model/model.go
Auto-merging lib/config/optionsconfiguration.go
CONFLICT (content): Merge conflict in lib/config/optionsconfiguration.go
Auto-merging lib/config/config_test.go
Auto-merging cmd/syncthing/main.go
CONFLICT (content): Merge conflict in cmd/syncthing/main.go
Squash commit -- not updating HEAD
Automatic merge failed; fix conflicts and then commit the result.

```

**Comment by calmh** [comment]
Okay then

**Comment by calmh** [comment]
@st-review merge nao?

**Comment by st-review** [comment]
:ok_hand: Merged as c4ba580cbb64521e6ca66b4f6bbbb3b3b14a39d7. Thanks, @calmh!

**Pull request closed without merging**